src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -248,145 +198,16 @@ export const Deposit: React.FC<DepositProps> = ({
-          const customTxInput = await createBuildCustomTxInput({
-            accountNumber,
-            adapter,
-            data,
-            value: '0', // this is not a token send, but a smart contract call so we don't send anything here, THOR router does
-            to: saversRouterContractAddress!,
-            wallet,
-          })
-
-          const fees = (await estimateFees({
-            accountId,
-            contractAddress: undefined,
-            assetId,
-            sendMax: false,
-            amountCryptoPrecision: '0',
-            to: customTxInput.to,
-            from: fromAccountId(accountId).account,
-            memo: customTxInput.data,
-          })) as FeeDataEstimate<KnownChainIds.EthereumMainnet>
-
-          const fastFeeCryptoBaseUnit = fees.fast.txFee
-
-          const fastFeeCryptoPrecision = fromBaseUnit(fastFeeCryptoBaseUnit, feeAsset.precision)
-
-          return fastFeeCryptoPrecision
-        }
-
-        const adapter = chainAdapters.get(chainId)!
-        const getFeeDataInput = {
-          to: quote.inbound_address,
-          value: amountCryptoBaseUnit,
-          chainSpecific: {
-            pubkey: userAddress,
-            from: fromAccountId(accountId).account,
-          },
-          sendMax: Boolean(state?.deposit.sendMax),
-        }
-        const fastFeeCryptoBaseUnit = (await adapter.getFeeData(getFeeDataInput)).fast.txFee
-        const fastFeeCryptoPrecision = fromBaseUnit(fastFeeCryptoBaseUnit, asset.precision)
-
-        return fastFeeCryptoPrecision
-      } catch (error) {
-        console.error(error)
-        toast({
-          position: 'top-right',
-          description: translate('common.somethingWentWrongBody'),
-          title: translate('common.somethingWentWrong'),
-          status: 'error',
-        })
-      }
-    },
-    [
-      userAddress,
-      assetReference,
-      accountId,
-      opportunityData,
-      accountNumber,
-      wallet,
-      feeAsset,
-      isThorchainSaversDepositQuoteLoading,
-      asset,
-      isThorchainSaversDepositQuoteError,
-      thorchainSaversDepositQuoteError?.message,
-      thorchainSaversDepositQuote,
-      isTokenDeposit,
-      isApprovalRequired,
-      chainId,
-      state?.deposit.sendMax,
-      saversRouterContractAddress,
+  const { estimatedFeesData, isEstimatedFeesDataLoading, outboundFeeCryptoBaseUnit } =
+    useSendThorTx({
       assetId,
-      toast,
-      translate,
-    ],
-  )
-
-  const {
-    data: estimatedFeesData,
-    isLoading: isEstimatedFeesDataLoading,
-    isSuccess: isEstimatedFeesDataSuccess,
-  } = useGetEstimatedFeesQuery({
-    amountCryptoPrecision: inputValues?.cryptoAmount ?? '0',
-    assetId,
-    to: thorchainSaversDepositQuote?.inbound_address ?? '',
-    sendMax: false,
-    accountId: accountId ?? '',
-    contractAddress: undefined,
-    enabled: Boolean(thorchainSaversDepositQuote && accountId && isUtxoChainId(asset.chainId)),
-  })
+      accountId: accountId ?? null,
```

Comment:
Non-blocking observation: this mix of `string`, `string | undefined` and `string | null` in our codebase is annoying and unnecessary

Reply:
100% agree! We've actually improved things out in this PR but still far from perfect. At least things are now encapsulated in a sane reused hook which expects `null` 🙏🏽

Reply:
I made the same comment and would like to vote on using `undefined` across the board unless `null` is used by 3rd party library. Thoughts?

Reply:
I'm strongly opinionated on `null` but happy to go with what we all decide to go with!

Reply:
Lets discuss this on this weeks team call